### PR TITLE
Created react-renderer and migrated depending classes to react

### PR DESCRIPTION
**Pull request by jbicker**
Depends on https://github.com/theia-ide/theia/pull/2291. So better review that one first and then rebase this.

### CODE REVIEW

File: packages/core/src/browser/widgets/react-renderer.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+export class ReactRenderer {
```
Comment (akosyakov) [inline packages/core/src/browser/widgets/react-renderer.tsx]: `extends Disposable`

File: packages/filesystem/src/browser/location/location-renderer.tsx [hunk 2]
```diff
@@ -39,24 +39,20 @@ export class LocationListRenderer extends VirtualRenderer {
         }
     }
 
-    protected doRender(): h.Child {
+    protected readonly locationChanged = (e: React.ChangeEvent<HTMLSelectElement>) => this.onLocationChanged(e);
```
Comment (akosyakov) [inline packages/filesystem/src/browser/location/location-renderer.tsx]: method names should be verbs, i.e `handleLocationChanged`

File: packages/filesystem/src/browser/location/location-renderer.tsx [hunk 3]
```diff
@@ -14,14 +14,14 @@
...
-import { VirtualRenderer } from "@theia/core/lib/browser";
 import URI from "@theia/core/lib/common/uri";
 import { LocationService } from "./location-service";
+import { ReactRenderer } from "@theia/core/lib/browser/widgets/react-renderer";
+import * as React from 'react';
 
 export const LOCATION_LIST_CLASS = 'theia-LocationList';
 
-export class LocationListRenderer extends VirtualRenderer {
+export class LocationListRenderer extends ReactRenderer {
```
Comment (akosyakov) [inline packages/filesystem/src/browser/location/location-renderer.tsx]: please make sure that `LocationListRenderer` is disposed together with `FileDialog`

Comment (jbicker) [reply packages/filesystem/src/browser/location/location-renderer.tsx]: https://github.com/theia-ide/theia/pull/2301/files#diff-f62deb5fbabbd0bb37a733cc83c2ba8fR86

**Pull request merged**